File: src/app/service/rest/user/user.service.ts
import {Injectable} from '@angular/core';
import {RestService} from '../rest.service';
import {UserBean} from '../../../obj/bean/user.bean';

@Injectable({
  providedIn: 'root',
})
export class UserService {

  constructor(private readonly restService: RestService) {
  }

  public isUserExisting$() {
    return this.restService.get$<boolean>('/auth/check');
  }

  public authenticate$(username: string, password: string) {
    return this.restService.post$<UserBean>('/auth/authenticate', {
      username,
      password,
    });
  }

  public createUser$(username: string, password: string) {
    return this.restService.post$<UserBean>('/auth/register', {
      username,
      password,
    });
  }

}

File: backend/src/main/java/dev/omel/service/UserService.java
package dev.omel.service;

import dev.omel.bean.UserBean;

public interface UserService {

  boolean isUserExisting();

  UserBean createUser(String username, String password);

  UserBean updateUser(Long id, UserBean user);

  UserBean getUserById(Long id);

  UserBean authenticate(String username, String password);
}
